src/Microsoft.OData.Core/UriParser/Parsers/SelectExpandParser.cs

```diff
@@ -81,7 +81,7 @@ internal sealed class SelectExpandParser
 
             // Sets up our lexer. We don't turn useSemicolonDelimiter on since the parsing code for expand options,
             // which is the only thing that needs it, is in a different class that uses it's own lexer.
-            this.lexer = clauseToParse != null ? new ExpressionLexer(clauseToParse, false /*moveToFirstToken*/, false /*useSemicolonDelimiter*/) : null;
+            this.lexer = clauseToParse != null ? new ExpressionLexer(clauseToParse, false, false, false, false) : null;

```

Comment:
>false [](http://example.com/codeflow?start=105&length=5)

so, only in select and expand, it's false, in others it's true?

We don't allow: $filter=true eq true, if one of the 'true' is for the property?

Reply:
There's a lot going on here. For `$filter=true eq true`, that's the same as just saying `$filter=true` or not providing the filter at all. We interpret the "true" literal as a boolean expression, which as far as I can tell is in accordance with the standard. If there is a property named "true", the above is still accurate, but the client can disambiguate that they are referring to the property by using `$it`, as in `$filter=$it/true eq true`. The reality is that we also have a bug in that case as well. I talked with Mike yesterday and we decided that we would address the `$filter` issue separately, since it implies a larger change (specifically, `ExpressionLexer` applies semantics, but many query options require on a syntax-only pass first; this syntax-only pass using `ExpressionLexer` has issues across all of the query option implementations).